src/Adapter/Tornado/EventLoop.php

```diff
@@ -46,12 +49,19 @@ function (\Throwable $throwable) use (&$finalAction, &$promiseIsPending) {
             return count($this->tasks) !== 0;
         };
 
+        if (method_exists($promise, 'isCancelled') && $promise->isCancelled()) {
+            throw new CancelledException('cancelled wait');
+        }
```

Comment:
🚫 No 
You can use `Internal\PendingPromise::toHandledPromise($promise)` to safely cast the promise interface to a concrete internal promise… if needed :)